app/apps/handlers.js

```diff
@@ -1,122 +1,100 @@
-const ensureLoggedIn = require('connect-ensure-login').ensureLoggedIn;
+const { ensureLoggedIn } = require('connect-ensure-login');
+const { App, Bucket, User } = require('../api-client');
 
-const api = require('../api-client');
-const routes = require('../routes');
 
-
-exports.new_app = [
+exports.new = [
   ensureLoggedIn('/login'),
-  function (req, res, next) {
-    api.list_buckets()
-      .then((buckets_response) => {
-        const template_args = {
+  (req, res, next) => {
+
+    Bucket.list()
+
+      .then((buckets) => {
+        res.render('apps/new.html', {
           prefix: process.env.ENV + '-',
-          buckets: buckets_response.results,
-        };
-        res.render('apps/new.html', template_args);
+          buckets: buckets,
+        });
       })
+
       .catch(next);
   }
 ];
 
 
-exports.create_app = [
+exports.create = [
   ensureLoggedIn('/login'),
-  function (req, res, next) {
-    const app = {
+  (req, res, next) => {
+
+    const { url_for } = require('../routes');
```

Comment:
If we're using `url_for()` in different places we may want to move this require at the top of the file.

Reply:
`routes` cannot be imported at the top of a `handlers` module, because it depends on all the handlers being loaded to work.

Reply:
The reason we had trouble doing `const { url_for } = require('../routes')` in a handler is because there is a cyclic dependency: `routes` requires all the handlers, which require `routes`.

Reply:
Oohhh, so this was the problem with this require, it was really weird.